src/Hangfire.Mongo/Migration/Steps/Version09/00_CreateSignalCollection.cs

```diff
@@ -17,13 +19,25 @@ public bool Execute(IMongoDatabase database, MongoStorageOptions storageOptions,
 
             database.DropCollection(name);
 
-            var createOptions = new CreateCollectionOptions
+            if (storageOptions.UseForCosmosMongoApi)
             {
-                Capped = true,
-                MaxSize = 1000000,
-                MaxDocuments = 1000
-            };
-            database.CreateCollection(name, createOptions);
+                database.CreateCollection(name);
+                var collection = database.GetCollection<BsonDocument>(name);
+                var options = new CreateIndexOptions { ExpireAfter = TimeSpan.FromHours(storageOptions.CosmosHourlyTtl) };
+                var field = new StringFieldDefinition<BsonDocument>("_ts");
+                var indexDefinition = new IndexKeysDefinitionBuilder<BsonDocument>().Ascending(field);
+                collection.Indexes.CreateOne(indexDefinition, options);
+            }
+            else
+            {
+                var createOptions = new CreateCollectionOptions
+                {
+                    Capped = true,
+                    MaxSize = 1000000,
+                    MaxDocuments = 1000
+                };
+                database.CreateCollection(name, createOptions);
+            }
```

Comment:
if cosmos db doesnt support capped collections, then you could just do an early return. no need to create the "notifications" collection.
if cosmos db dont return  "MongoNotificationObserver" in "MongoStorage class in method "GetComponents"

Reply:
Use a TTL collection to simulate a Mongo capped collection